cheta/fetch.py

```diff
@@ -538,11 +538,16 @@ def dtype(self):
 
     def __repr__(self):
         attrs = [self.__class__.__name__, self.MSID]
+        # Try to get dtype name, but handle objects without .dtype like Quat
```

Comment:
This could be handled upstream by fixing the `dtype` property:
```
    @property
    def dtype(self) -> np.dtype:
        return self.vals.dtype if hasattr(self.vals, "dtype") else np.dtype("object")
```
This ensures that the dtype property works and returns a real numpy dtype. For `vals` that are some non-ndarray class, it seems like the `object` dtype is a good proxy. This also matches what is done in astropy tables for mixin columns (the `info.dtype` of such a column is `dtype('object')`).